lib/chef/provider/package/aix.rb

```diff
@@ -88,18 +85,20 @@ def load_current_resource
 
         def candidate_version
           return @candidate_version if @candidate_version
-          ret = shell_out_with_timeout("installp -L -d #{@new_resource.source}")
-          ret.stdout.each_line do |line|
-            case line
-            when /\w:#{Regexp.escape(@new_resource.package_name)}:(.*)/
-              fields = line.split(":")
-              @candidate_version = fields[2]
-              @new_resource.version(fields[2])
-              Chef::Log.debug("#{@new_resource} setting install candidate version to #{@candidate_version}")
+          if package_source_found?
+            ret = shell_out_with_timeout("installp -L -d #{@new_resource.source}")
```

Comment:
I know this is existing code, but in the spirit of leaving things better than we found them: The code from 89-98 looks really similar to the code from 53-62. Both appear to be extracting the version from a given package on disk. However, it isn't clear why this regex is different from those used above.  Could we consolidate this to a single function, perhaps with some comments?